src/App.tsx

```diff
@@ -1,24 +1,153 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { createTodo, getTodos, deleteTodo } from './api/todos';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { Main } from './components/Main/Main';
+import { ErrorMessages } from './components/ErrorMessages/ErrorMessages';
 
-const USER_ID = 0;
+const USER_ID = 10548;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [input, setInput] = useState('');
+  const [status, setStatus] = useState('all');
+  const [disableInput, setDisableInput] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>();
+  const [hasError, setHasError] = useState(false);
+  const [typeError, setTypeError] = useState('');
+
+  async function loadedTodos(f: (USER_ID: number) => Promise<Todo[]>) {
+    try {
+      const result = await f(USER_ID);
+
+      setTodos(result);
+      setHasError(false);
+    } catch (error) {
+      setTypeError('Unable to load a todo');
+      setHasError(true);
+    }
+  }
+
+  useEffect(() => {
+    loadedTodos(getTodos);
+  }, []);
+
+  const handleChangeInput = (event : React.ChangeEvent<HTMLInputElement>) => {
+    setInput(event.target.value);
+  };
+
+  const handleStatus = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    setStatus(event.currentTarget.type);
+  };
+
+  const handleAddTodo = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setDisableInput(true);
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: input,
+      completed: false,
+    });
+
+    setInput('');
+    try {
+      await createTodo(USER_ID, {
+        title: input,
+        userId: USER_ID,
+        completed: false,
+      });
+
+      await loadedTodos(getTodos);
+      setTempTodo(null);
+      setDisableInput(false);
+      setHasError(false);
+    } catch (error) {
+      setTypeError('Unable to add a todo');
+      setHasError(true);
+    }
+  };
+
+  const handleRemoveTodo = async (id: number) => {
+    try {
+      await deleteTodo(id);
+      loadedTodos(getTodos);
```

Comment:
no need to load todos again from server
after success response filter todos by id and update state

Reply:
![Screenshot from 2023-05-29 13-03-11](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/33029864/5375eb43-62b4-47ec-ba83-261d7b271178)